### PR TITLE
Switch to GitHub Actions

### PR DESCRIPTION
**Pull request by gjtorikian**

### CODE REVIEW

**Comment by gjtorikian** [comment]
@gogainda It seems that `truffleruby` is failing to compile Nokogiri. I've removed it from this PR but you're more than welcome to open a new PR against the `v3` branch to get it working.

**Comment by gogainda** [comment]
@eregon I just double checked nokoigiri locally, it's installing correctly. Not sure why it's failing in GitHub Actions

**Comment by eregon** [comment]
See the workflow step in https://github.com/oracle/truffleruby/issues/62#issue-207675764
TruffleRuby needs a system libxml2-dev libxslt-dev to be installed, and that's not the default on GitHub Actions (but it was on TravisCI).

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - uses: actions/checkout@v2
+      - name: Set up Ruby ${{ matrix.ruby-version }}
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+          bundler-cache: true
+      - name: Install dependencies
+        run: |
+          bundle config set --local path '../../vendor/bundle'
+          bundle install
```
Comment (eregon) [inline .github/workflows/ci.yml:39]: Unneeded, and most likely breaks bundler caching, see https://github.com/ruby/setup-ruby/#caching-bundle-install-automatically

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  test:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.0
+          bundler-cache: true
+      - run: bundle install
```
Comment (eregon) [inline .github/workflows/lint.yml:20]: redundant with `bundler-cache: true` above

**Comment by gogainda** [comment]
@eregon thanks, I will open a new PR to fix it

**Comment by gogainda** [comment]
https://github.com/gjtorikian/html-pipeline/pull/348

**Pull request merged**